usaspending_api/references/management/commands/population_data_loaders/loaders.py

```diff
@@ -0,0 +1,148 @@
+        self._columns_mapper = column_mapper
+        self._logger = logger
+
+    def drop_temp_tables(self):
+        self._logger.info(f"Dropping temp table {self.TEMP_TABLE_NAME}")
+        with connection.cursor() as cursor:
+            cursor.execute(f"DROP TABLE IF EXISTS {self.TEMP_TABLE_NAME}")
+
+    def create_tables(self, columns):
+        self._logger.info(f"Creating temp table {self.TEMP_TABLE_NAME}")
+        with connection.cursor() as cursor:
+            cursor.execute(f"DROP TABLE IF EXISTS {self.TEMP_TABLE_NAME}")
+            cursor.execute(
+                self.TEMP_TABLE_SQL.format(table=self.TEMP_TABLE_NAME, columns=",".join([f"{c} TEXT" for c in columns]))
+            )
+
+    def load_data(self, data, model=None):
+        self._logger.info(f"Attempting to load {len(data)} records into {model.__name__}")
+        model.objects.all().delete()
+        model.objects.bulk_create(
+            [model(**{col: row[csv] for csv, col in self._columns_mapper.items()}) for row in data]
+        )
+        self._logger.info("Success? Please Verify")
+
+    def cleanup(self):
+        self.drop_temp_tables()
+
+
+class DistrictPopulationLoader(GenericPopulationLoader):
+    """A concrete implemention of the generic population loader. Suitable for loading
+    district population data.
+    """
+
+    def load_data(self, data, model=None):
+        model = PopCongressionalDistrict
+        super().load_data(data, model)
+
+
+class CountyPopulationLoader(GenericPopulationLoader):
+    """A concrete implemention of the generic population loader. Suitable for loading
+    county population data.
+    """
+
+    def load_data(self, data, model=None):
+        model = PopCounty
+        super().load_data(data, model)
+
+    def cleanup(self):
+        self.drop_temp_tables()
+
+
+class CountryPopulationLoader(GenericPopulationLoader):
+    """A concrete implemention of the generic population loader. Suitable for loading
+    country population data.
+    """
+
+    ERROR_THRESHOLD = 0.5
+
+    def load_data(self, data, model=None):
+        model = RefCountryCode
+        # Ensuring that the update will result under our accepted error threshold
+        total_countries = model.objects.all().distinct("country_code")
+        if float(len(data)) / float(len(total_countries)) < self.ERROR_THRESHOLD:
+            raise RuntimeError(
+                "The provided data contains less than 50% of the known countries. "
+                "We require at least 50% of the known countries to be present in the file to load. "
+                f"Data had {len(data)} countries when there are {len(total_countries)} countries in {model.__name__}."
+            )
+        self._logger.info(f"Attempting to load {len(data)} records into {model.__name__}")
+
+        for row in data:
+            record = model.objects.get(country_code=row["country_code"])
+            record.latest_population = row["population"]
+            record.save()
```

Comment:
This will update the `update_date` field on the record every time a record with a population is read from the file. It would be nice if this only updated that date field whenever the population of a record changes.

Reply:
addressed. Please see changes.